### PR TITLE
Implement IPv6 route programming

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

File: vpc/tool/container2/setup_container_linux.go [hunk 1]
```diff
@@ -254,52 +249,169 @@ func configureLink(ctx context.Context, nsHandle *netlink.Handle, link netlink.L
...
+	m := &ndp.RouterSolicitation{
+		Options: []ndp.Option{
+			&ndp.LinkLayerAddress{
+				Direction: ndp.Source,
+				Addr:      networkInterface.HardwareAddr,
+			},
+		},
+	}
+
+	/* Poll for the router */
```
Comment (solarkennedy) [inline vpc/tool/container2/setup_container_linux.go:337]: What do you think about a function for this called waitForIPV6NDPAdvertisement ?

File: vpc/tool/container2/setup_container_linux.go [hunk 2]
```diff
@@ -254,52 +249,169 @@ func configureLink(ctx context.Context, nsHandle *netlink.Handle, link netlink.L
...
+		}
+		if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+			logger.G(ctx).Debug("I/O read timeout")
+			continue
+		}
+		return fmt.Errorf("Could not read router soliticiation response / router advertisement: %w", err)
 	}
+
+	for _, route := range allocation.Routes {
+		if route.Family != vpcapi.AssignIPResponseV3_Route_IPv6 {
```
Comment (solarkennedy) [inline vpc/tool/container2/setup_container_linux.go:384]: This confuses me.

Are we going to get `allocation.Routes` that are ipv6, but we won't want them? Is this like a feature flag?

**Review by solarkennedy — APPROVED** [review]

File: vpc/tool/container2/setup_container_linux.go (hunk 2 above)
Comment (sargun) [reply vpc/tool/container2/setup_container_linux.go:384]: hrm? this is equivalent to putting the for loop in a route.family == IPv6 {
...add route..
}

**Review by alekar — APPROVED** [review]

File: vpc/tool/container2/setup_container_linux.go (hunk 2 above)
Comment (solarkennedy) [reply vpc/tool/container2/setup_container_linux.go:384]: Oh ok. I couldn't tell what `AssignIPResponseV3_Route_IPv6` is. Did it just get added to the api definitions?

Comment (sargun) [reply vpc/tool/container2/setup_container_linux.go:384]: yep.

Comment (sargun) [reply vpc/tool/container2/setup_container_linux.go:384]: it's how enums work in protobuf 🤷

File: vpc/tool/container2/setup_container_linux.go [hunk 3]
```diff
@@ -254,52 +249,169 @@ func configureLink(ctx context.Context, nsHandle *netlink.Handle, link netlink.L
...
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// Amazon only gives out /128s
+	new6IP := net.IPNet{IP: net.ParseIP(allocation.IPV6Address.Address.Address), Mask: net.CIDRMask(128, 128)}
+	new6Addr := netlink.Addr{
+		// TODO (Sargun): Check IP Mask setting.
+		IPNet: &new6IP,
+		Flags: unix.IFA_F_PERMANENT | unix.IFA_F_NODAD,
```
Comment (rgulewich) [inline vpc/tool/container2/setup_container_linux.go:267]: Why `IFA_F_NODAD`?  Might be worth a comment.

**Review by rgulewich — APPROVED** [review]

**Comment by sargun** [comment]
@DonavanFritz *nudge*

**Pull request merged**